### DISCUSSION TITLE
Let the custom titlebar act like the system titlebar?

### DISCUSSION THREAD

**Discussion by timwehrle** [tauri-apps/tauri#10221] [Q&A]
So I've been wondering about if I can get the custom titlebar to act like the system titlebar in terms of resizing. I guess the custom titlebar is always slower when I resize the window because it gets rendered in the webview. Is there a way to achieve the same effect like the system titlebar?

With system titlebar
https://github.com/tauri-apps/tauri/assets/115550741/c68c6fef-8408-4d38-b642-e83d5675be97

With custom titlebar
https://github.com/tauri-apps/tauri/assets/115550741/a2fd076b-4970-48d6-abc2-d85051bbfc3f

**Comment by FabianLars** [comment]
With html/css based titlebars this can really only be fixed by fixing resizing itself which has multiple gh issues like https://github.com/tauri-apps/tauri/issues/6322 for example.

The alternative to this would be a native "custom" titlebar which a) is pretty difficult in general and b) basically incompatible with tauri or webview2 depending on the approach (there should be a few discussions on this and wry's github and our discord).

**Reply by timwehrle** [reply]
And what is the preferred way to solve this problem? Make users use the native bar or just use all the workarounds until something is finally fixed?

**Reply by FabianLars** [reply]
Depends on your app/needs. If the native titlebar works for you, then i'd recommend sticking to it. If not, try the workarounds or help us fixing the issues :)

**Reply by timwehrle** [reply]
I mean, I'd like to help, but your response kind of made it seem like it wasn't a Tauri problem, but a webview problem, didn't it? Probably helping with the resizing problem would be good?

**Reply by FabianLars** [reply]
It's a mix of both/all.
- Chromium's resizing itself is already pretty bad, at least if hardware acceleration is enabled - On that note, if your app isn't too graphics intensive, this may be an option for you? Software rendered resizing _should_ be as smooth as it can get. You can do this by setting `--disable-gpu` in the `additionalBrowserArgs` config.
- webview2' resizing somehow seems to be worse than chromium, talking with their team wasn't really fruitful though.
- wry's resizing may also be a tad bit slower than raw webview2, not sure
- Then the issue i linked above comes into play, for some reason resizing is _much_ worse in tauri compared to wry so something in tauri's event loop is pretty messed up. Fixing the actual root causes needs deep tauri knowledge, but integrated workarounds in the spirit of https://github.com/tauri-apps/tauri/issues/6322#issuecomment-1872491965 / https://github.com/tauri-apps/tauri/pull/8801 may be possible for normal contributors too.

This isn't reallyyy my area of expertise anymore either :/

**Reply by timwehrle** [reply]
Yea, thanks for the overview. I'll see what I can do or if I can do something lol.
